plugins/cpp/service/src/filediagram.cpp

```diff
@@ -415,28 +415,47 @@ std::vector<util::Graph::Node> FileDiagram::getImplementedFiles(
   const util::Graph::Node& node_,
   bool reverse_)
 {
-  std::vector<util::Graph::Node> implements;
-
-  _transaction([&, this]{
-    EdgeResult res = _db->query<model::CppEdge>(
-      (reverse_
-         ? EdgeQuery::to->domainId
-         : EdgeQuery::from->domainId) == node_ &&
-      EdgeQuery::from->domainId != EdgeQuery::to->domainId &&
-      EdgeQuery::type == model::CppEdge::IMPLEMENT);
+  std::vector<core::FileId> implements;
+  _transaction([&, this]
+  {
+    auto contained = _db->query<model::File>(
+      odb::query<model::File>::parent == std::stoull(node_) &&
+      odb::query<model::File>::type != model::File::DIRECTORY_TYPE
+    );
+    std::unordered_set<core::FileId> used;
 
-    for (const model::CppEdge& edge : res)
+    for (const model::File &file : contained)
     {
-      core::FileId fileId = reverse_ ? edge.from->domainId : edge.to->domainId;
-
-      core::FileInfo fileInfo;
-      _projectHandler.getFileInfo(fileInfo, fileId);
+      auto files = getProvidedFileIds(graph_, std::to_string(file.id), reverse_);
+      used.insert(files.begin(), files.end());
+    }
+    for (const core::FileId& fileId : used)
+    {
+      auto files = _db->query<model::File>(
+        odb::query<model::File>::id == std::stoull(fileId)
```

Comment:
Actually, if this is an ID, then can this query contain multiple results? Perhaps `load()` should be used here instead.

Reply:
Fixed in 92e2fcf9a650d61a4b2a65f8c8805e107cd52b33